lib/core/src/sdk.rs

```diff
@@ -4001,13 +4058,43 @@ impl LiquidSdk {
 
     /// Synchronizes the local state with the mempool and onchain data.
     pub async fn sync(&self, partial_sync: bool) -> SdkResult<()> {
-        self.sync_inner(partial_sync, None).await
+        let liquid_tip = self.liquid_chain_service.tip().await?;
+
+        let last_bitcoin_tip = self.get_info().await?.blockchain_info.bitcoin_tip;
+        let last_chain_tips = ChainTips {
+            liquid_tip,
+            bitcoin_tip: Some(last_bitcoin_tip),
+        };
+        let recoverable_swaps = self
+            .get_monitored_swaps_list(partial_sync, true, last_chain_tips)
+            .await?;
+
+        let chain_tips = ChainTips {
+            liquid_tip,
+            bitcoin_tip: {
+                if recoverable_swaps
+                    .iter()
+                    .any(|s| matches!(s, Swap::Chain(_)))
+                    .not()
+                {
+                    info!("No chain swaps being monitored, skipping bitcoin tip fetch");
+                    None
+                } else {
+                    self.bitcoin_chain_service.tip().await.ok()
+                }
+            },
+        };
```

Comment:
Eventually the sync_inner needs two things: recoverable_swaps and chain_tips.
The above code block gets this info and we also getting this info in the `track_new_blocks` function.
Do you think we can create one function to fetch this info that can be reused in these two places?
Something like `get_sync_context` ? It will help get out the small details out of the main flow.